embassy-usb/src/lib.rs

```diff
@@ -194,7 +201,7 @@ impl<'d, D: Driver<'d>> UsbDevice<'d, D> {
             }
         }
 
-        while !self.inner.suspended {
+        while !self.inner.suspended && self.inner.power_available {
```

Comment:
The behavior of this method is a bit inconsistent. If you call it when `!self.inner.power_available`, it will wait for power, but if power becomes unavailable while it's running it will return. I think it should probably return only when the bus enters the suspended state and handle power changes without returning.

Reply:
That's not valid per the USB spec. If power is interrupted the device state is reset and you have to start over.

Reply:
Understood. However, those events come from separate sources, so it is technically possible.